addon-test-support/@ember/test-helpers/dom/blur.ts

```diff
@@ -24,14 +29,17 @@ export function __blur__(element: HTMLElement | Element | Document | SVGElement)
 
   // makes `document.activeElement` be `body`.
   // If the browser is focused, it also fires a blur event
-  element.blur();
+  if (!relatedTarget) {
```

Comment:
I might miss something, but it looks like the `__blur__(` would be no-op, in case if there is no `relatedTarget` and browser is focused?

Reply:
`relatedTarget` defaults to `null` if it is not provided, meaning that `element.blur();` will execute if it's not set by way of the `!` operator. I can make this a stricter comparison so that it's clearer, for example:
`if (relatedTarget === null) {`

Reply:
Ya, I'd vaguely prefer to check directly against `null`. I _might_ even tweak to clarify with a variable 🤔. Something like:

```suggestion
  let relatedTargetProvided = relatedTarget !== null;
  if (!relatedTargetProvided) {
```

Reply:
I'm not 100% sure that this case was what @ro0gr was mentioning, but I do think we still have a logic problem here @drewlee. When `relatedElement` is passed as a non-null value, **and** the browser is in focus we never fire `blur` event.

Reply:
@rwjblue this is intentional. The idea is that if the browser is in focus, and `__blur__` is triggered from `__focus__`, we shouldn't need to do anything. Based on my findings, the browser automatically fires native blur when `element.focus()` is called with the expected event properties (including `relatedTarget`). Otherwise, blur and focusout end up executing multiple times, which I believe is also one of the causes of the issue. I think the initial tests were masking this behavior as the condition never executed due to `activeElement` being the body. Let me set something up to better demonstrated what I mean, at least for my own sanity. It could be that the check to trigger `__blur__` needs to be modified, or I'm somehow getting trolled by the associated tests.

Reply:
OK, that generally makes sense. We need to change this around so that the `focus` side deals with that nuanced detail because this code in `__blur__` should make the assumption that it is called from `focus`.

I think from the `__blur__` perspective, we should do the snippet I mentioned in a separate comment, then update `focus` to handle the "don't call __blur__ if you don't need to" case.

Reply:
Makes sense. I've incorporated this in my latest commit.